src/main/java/com/gmail/inverseconduit/bot/InteractionBot.java

```diff
@@ -26,13 +26,12 @@ public InteractionBot(ChatInterface chatInterface) {
 
     @Override
     public void start() {
-        executor.scheduleAtFixedRate(this::processInteractions, 200, 700, TimeUnit.MILLISECONDS);
+        // asynchronously enqueue the processing by blocking supplier
+        processingThread.submit(this::processInteractions);
```

Comment:
There is a blocking queue. I changed the blocking supplier from `poll()` to `take()`, which block until an element becomes available. These elements are then streamed infinitely and enqueued into a separate thread to be processed.